File: packages/elixir-sdk/test/adapters/knowledge/query_test.exs
defmodule TreeDxSdk.QueryAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Query.read_file(client, "repo/a", %{})
    TreeDxSdk.Query.list_paths(client, "repo/a", %{})
    TreeDxSdk.Query.search_files(client, "repo/a", %{})
    TreeDxSdk.Query.repository(client, "repo/a", %{})

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :post and &1.path == "/api/v1/repos/repo%2Fa/query")
           )
  end
end

File: packages/ts-sdk/test/adapters/knowledge/query.test.ts
import { describe, expect, it } from 'vitest';
import { QueryAdapter } from '../../../src/treedx/index.js';
import { MockTransport } from '../mock.js';

describe('QueryAdapter', () => {
  it('constructs query endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new QueryAdapter({ transport });
    await adapter.readFile('repo', {});
    await adapter.listPaths('repo', {});
    await adapter.searchFiles('repo', {});
    await adapter.repository('repo', {});
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'POST /api/v1/repos/repo/files/read',
      'POST /api/v1/repos/repo/paths/list',
      'POST /api/v1/repos/repo/files/search',
      'POST /api/v1/repos/repo/query'
    ]);
  });
});
